pkg/common/cryptoutil/keys.go

```diff
@@ -48,3 +52,23 @@ func KeyMatches(privateKey crypto.PrivateKey, publicKey crypto.PublicKey) (bool,
 		return false, fmt.Errorf("unsupported private key type %T", privateKey)
 	}
 }
+
+func GenerateSigner(keyType keymanagerv0.KeyType) (signer crypto.Signer, err error) {
```

Comment:
We should use types and constants from the `pkg/server/plugin/keymanager` package instead of those from the v0 version of the keymanager plugin interface.

Reply:
In fact, I wonder if we shouldn't add this function as a method on the KeyType type.

```
func (keyType KeyType) Generate() (crypto.Signer, error)
```

Reply:
Thanks, having the function on `KeyType` simplifies things quite a lot.